domain/src/main/java/org/cryptomator/domain/UnverifiedVaultConfig.kt

```diff
@@ -3,4 +3,19 @@ package org.cryptomator.domain
 import java.io.Serializable
 import java.net.URI
 
-open class UnverifiedVaultConfig(open val jwt: String, open val keyId: URI, open val vaultFormat: Int) : Serializable
+open class UnverifiedVaultConfig(open val jwt: String, open val keyId: URI, open val vaultFormat: Int) : Serializable {
+	fun keyLoadingStrategy(): KeyLoadingStrategy = KeyLoadingStrategy.fromKeyId(keyId)
+}
+
+enum class KeyLoadingStrategy(private val prefix: String) {
+	MASTERKEY("masterkey+"),
+	HUB("hub+http");
+
+	companion object {
+		fun fromKeyId(keyId: URI): KeyLoadingStrategy {
+			val keyIdStr = keyId.toString()
+			return entries.firstOrNull { keyIdStr.startsWith(it.prefix) }
+				?: throw IllegalArgumentException("Unsupported keyId prefix: $keyId")
```

Comment:
Just to decide it once and for all: Do we want to stay with `startsWith` or is getting the scheme and then performing a check for equality the better approach here?

Reply:
I prefer `startsWith` because if we want to use `equals` the entries have to support multiple values e.g. `HUB(listOf("hub+http", "hub+https"))` and I do not think that this makes it really cleaner.